### PR TITLE
[Documentation] fix link to edit-post documentation

### LINKED ISSUE #12332: Link in plugins package README leads to Github 404 when clicked on in npmjs.com
**Describe the bug**
In the readme for the plugins package, there is the text "See the edit-post module documentation for available components."  The words "edit-post" link to here:

https://github.com/WordPress/gutenberg/blob/edit-post/

As of today November 26th, 2018, that is a 404.

**To Reproduce**
1. Go to https://www.npmjs.com/package/@wordpress/plugins
2. Click on the link to edit-post docs
3. See 404

**Expected behavior**

I'm also confused -- this may belong in a separate issue -- as I assume the link should go here: https://www.npmjs.com/package/@wordpress/edit-post but that says the component is for internal use only, but it appears to be necessary for sidebar plugins according to the plugins readme.

### PR DESCRIPTION
**Pull request by torounit**
## Description

Fixes #12332.

Fix link to edit-post documentation.

### CODE REVIEW

**Comment by chrisvanpatten** [comment]
@gziolo I’m open to ideas. Right now the parser on dot org requires a full relative path which also works on github, eg `../../../docs/path/to/file.md` (or whatever tha path is) where you have to traverse up to the base of the repo, and then back down to your file. 

Obviously that sucks from a UX perspective but also it does mean issues on GitHub.

For NPM readmes I think either the full URL is reasonable (I’d prefer linking to the GH version but the NPM version is fine too) but we need help from @dd32 (or someone else with bandwidth) to make sure the parser knows how to handle that and rewrite it to a handbook URL.

**Comment by gziolo** [comment]
It looks like npm takes `https://github.com/WordPress/gutenberg/tree/master/` as the current folder for docs. It seems like it reads it from `package.json` file. Maybe we should change some field there and it will magically work :)

**Comment by gziolo** [comment]
So I found related issues:
https://npm.community/t/relative-links-in-npm-readme-markdown-not-functional/1525/6

I can confirm that `https://github.com/WordPress/gutenberg/tree/master/` is going to be considered as the root folder. To make it work for both NPM and GitHub we can use the following:
`/packages/edit-post/README.md`

I'm wondering if that would also work for docs:
`/docs/designers-developers/key-concepts.md`

@dd32 - assuming that I'm correct, would it be possible to make it work this way for the handbook? It seems like it would make it much easier at least on the GitHub side of things.

File: packages/plugins/README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ This method takes two arguments:
       or an element (or function returning an element) if you choose to render your own SVG.
     - `render`: A component containing the UI elements to be rendered.
 
-See [the edit-post module documentation](../edit-post/) for available components.
+See [the edit-post module documentation](../packages-edit-post/) for available components.
```
Comment (gziolo) [inline packages/plugins/README.md]:
```suggestion
See [the edit-post module documentation](/packages/edit-post/) for available components.
```

It seems like it works properly when I try it in the preview mode.

**Comment by chrisvanpatten** [comment]
@gziolo I like the idea of `/docs/xyz` in theory, but does that work inside GitHub? Will GitHub just treat that as `https://github.com/docs/xyz` or correctly rewrite it? (I'll do a quick test unless someone chimes in)

**Comment by gziolo** [comment]
You need to try with the file edit feature on GitHub and preview mode.

**Comment by dd32** [comment]
> @dd32 - assuming that I'm correct, would it be possible to make it work this way for the handbook? It seems like it would make it much easier at least on the GitHub side of things.

@gziolo If i'm understanding correctly, you'd like to make links starting as `/docs/...` and `/packages/...` work the same way as the `../docs/` urls currently do? or should `/packages/..` go to a subpage? etc.

If i'm misunderstanding, an exact list of how each should be treated would be helpful, similar to this:
 * `../docs/(.+)$` => `https://wordpress.org/gutenberg/handbook/$1`

**Comment by gziolo** [comment]
All documentation pages for packages are grouped under: https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/

It looks like it will be a slightly different then. I will check also which different patterns we use. There should be also something custom for components from the components package. I will comment soon with the list of mappings.

**Comment by gziolo** [comment]
```json
{
	"title": "Packages",
	"slug": "packages",
	"markdown_source": "https://raw.githubusercontent.com/WordPress/gutenberg/master/docs/designers-developers/developers/packages.md",
	"parent": "developers"
},
{
	"title": "@wordpress/edit-post",
	"slug": "packages-edit-post",
	"markdown_source": "https://raw.githubusercontent.com/WordPress/gutenberg/master/packages/edit-post/README.md",
	"parent": "packages"
},
{
	"title": "Components",
	"slug": "components",
	"markdown_source": "https://raw.githubusercontent.com/WordPress/gutenberg/master/packages/components/README.md",
	"parent": "developers"
},
{
	"title": "HigherOrder",
	"slug": "higher-order",
	"markdown_source": "https://raw.githubusercontent.com/WordPress/gutenberg/master/packages/components/src/higher-order/README.md",
	"parent": "components"
},
```

I think we need to support 4 url patterns from the list extracted above. Turning them into absolute urls would end up with the following regular expressions for the handbook:

- `/docs/(.+)$` => `https://wordpress.org/gutenberg/handbook/$1`
- `/packages/components/README.md$` => `https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/`
- `/packages/components/(.+)/README.md$` => `https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/$1`
- `/packages/(.+)/README.md$` =. `https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-$1/`

**Comment by chrisvanpatten** [comment]
@gziolo @dd32 I think `/docs/(.*) → https://wordpress.org/gutenberg/handbook/$1` is 100% safe to implement.

For package and component documentation, let's target that separately… I'll try to devote an hour or so today/tomorrow to figuring out there's a cleaner approach with packages and components.

**Comment by dd32** [comment]
I've added support for these links in https://meta.trac.wordpress.org/changeset/7989

As adding the extra links formats as mentioned by @gziolo in https://github.com/WordPress/gutenberg/pull/12835#issuecomment-447819488 was easier to do at the same time, I've added them all - If they don't get used let me know and I'll pull them back out of the docs importer, but having them there doesn't negatively affect anything.

The old-style `../docs` links still get processed as well, so a hard-cutover isn't needed or anything like that.

**Comment by gziolo** [comment]
@dd32 I did some debugging with the following snippet:

```php
$markdown = preg_replace( '/(\[.*?\]\((\.\.\/)+.*?)((\/readme)?\.md)?(#.*)?\)/i', '$1$5)', $markdown );
$markdown = preg_replace( '@(\[.*?\])\(/packages/(.*?)/?(#.*)?\)@i', '$1(https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-$2/$3)', $markdown );
print $markdown;
```
It works perfectly fine for:
* `/packages/edit-post` -> https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-edit-post

However, it doesn't remove `README.md` as it happens for the existing paths relative to `docs/`

* `/packages/edit-post/REAMDE.md` -> `https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-components/README.md/`

**Review by gziolo — APPROVED** [review]
It looks good now. Thanks for opening this PR 👍

**Comment by gziolo** [comment]
I can confirm that the link in https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-plugins/ works as expected 🎉

**Comment by chrisvanpatten** [comment]
Thanks @gziolo!! 🎉

**Comment by dd32** [comment]
> @dd32 I did some debugging with the following snippet:
> However, it doesn't remove `README.md` as it happens for the existing paths relative to `docs/`
> 
> * `/packages/edit-post/REAMDE.md` -> `https://wordpress.org/gutenberg/handbook/designers-developers/developers/packages/packages-components/README.md/`

Ahh yes, I had forgotten that change was actually needed.. Fixed up in https://meta.trac.wordpress.org/changeset/7993

**Pull request merged**